controllers/nstemplateset/cluster_resources_test.go

```diff
@@ -906,361 +526,278 @@ func TestPromoteClusterResources(t *testing.T) {
-				HasResource("for-"+spacename, &quotav1.ClusterResourceQuota{},
-					WithLabel("toolchain.dev.openshift.com/templateref", "fail-clusterresources-abcde11"),
-					WithLabel("toolchain.dev.openshift.com/tier", "fail")).
-				HasResource(spacename+"-tekton-view", &rbacv1.ClusterRoleBinding{},
-					WithLabel("toolchain.dev.openshift.com/templateref", "fail-clusterresources-abcde11"),
-					WithLabel("toolchain.dev.openshift.com/tier", "fail"))
+				HasResource("for-"+spaceName, &quotav1.ClusterResourceQuota{}).
+				HasResource(spaceName+"-tekton-view", &rbacv1.ClusterRoleBinding{})
 		})
 
 		t.Run("fail to downgrade from advanced to basic tier", func(t *testing.T) {
 			// given
-			nsTmplSet := newNSTmplSet(namespaceName, spacename, "basic", withNamespaces("abcde11", "dev"))
-			crq := newClusterResourceQuota(spacename, "advanced")
-			manager, cl := prepareClusterResourcesManager(t, nsTmplSet, crq, crb)
+			previousTierTemplate, err := createTierTemplate(scheme.Codecs.UniversalDeserializer(),
+				test.CreateTemplate(
+					test.WithObjects(advancedCrq),
+					test.WithParams(spacename),
+				), "advanced", "clusterresources", "previousrevision")
+			require.NoError(t, err)
+			nsTmplSet := newNSTmplSet(namespaceName, spaceName, "basic", withNamespaces("abcde11", "dev"), withStatusClusterResourcesInTier("advanced", "previousrevision"))
+			crq := newClusterResourceQuota(spaceName, "advanced")
+			manager, cl := prepareClusterResourcesManager(t, nsTmplSet, crq, crb, previousTierTemplate)
 			cl.MockDelete = func(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
 				return fmt.Errorf("some error")
 			}
 
 			// when
-			updated, err := manager.ensure(ctx, nsTmplSet)
+			err = manager.ensure(ctx, nsTmplSet)
 
 			// then
 			require.Error(t, err)
-			assert.False(t, updated)
-			AssertThatNSTemplateSet(t, namespaceName, spacename, cl).
+			AssertThatNSTemplateSet(t, namespaceName, spaceName, cl).
 				HasFinalizer().
 				HasConditions(UpdateFailed(
-					"failed to delete an existing redundant cluster resource of name 'for-johnsmith' and gvk 'quota.openshift.io/v1, Kind=ClusterResourceQuota': some error"))
+					"failed to delete obsolete object 'for-johnsmith' of kind 'ClusterResourceQuota' in namespace '': some error"))
 			AssertThatCluster(t, cl).
-				HasResource("for-"+spacename, &quotav1.ClusterResourceQuota{},
-					WithLabel("toolchain.dev.openshift.com/templateref", "advanced-clusterresources-abcde11"),
-					WithLabel("toolchain.dev.openshift.com/tier", "advanced")).
-				HasResource(spacename+"-tekton-view", &rbacv1.ClusterRoleBinding{},
-					WithLabel("toolchain.dev.openshift.com/templateref", "advanced-clusterresources-abcde11"),
-					WithLabel("toolchain.dev.openshift.com/tier", "advanced"))
+				HasResource("for-"+spaceName, &quotav1.ClusterResourceQuota{}).
+				HasResource(spaceName+"-tekton-view", &rbacv1.ClusterRoleBinding{})
 		})
 	})
 }
 
 func TestUpdateClusterResources(t *testing.T) {
```

Comment:
I have a feeling that here is some duplication with the tests under `TestPromoteClusterResources`, but it can be optimized in the following PR